support/ccm.rb

```diff
@@ -854,13 +854,23 @@ def cluster_name
   def cassandra_version
     parse_version
     @cassandra_version ||= begin
-      version = @raw_version
-      if (@raw_version.start_with?('4.0') || @raw_version.start_with?('4.5') || @raw_version.start_with?('4.6')) and @dse
-        version = '2.0.17'
-      elsif (@raw_version.start_with?('4.7') || @raw_version.start_with?('4.8')) and @dse
-        version = '2.1.12'
+      if @dse
+        return case
+               when @raw_version =~ /4.[056].*/
+                 '2.0'
+               when @raw_version =~ /4.[78].*/
+                 '2.1'
+               when @raw_version =~ /5.0.*/
+                 '3.0'
+               when @raw_version =~ /5.1.[01]/
+                 "3.10"
+               when @raw_version =~ /5.1.*/
+                 "3.11"
+               else
+                 "4.0"
```

Comment:
The real point of this change: the old version only ran up through DSE 4.8.  The conversion to the case statement here was just an attempt to make the relationships between DSE and C* versions easier to grok on a simple read.